playbooks/roles/oraclejdk/tasks/main.yml

```diff
@@ -21,6 +21,18 @@
     url: "{{ oraclejdk_url }}"
     headers: 'Cookie:oraclelicense=accept-securebackup-cookie'
     dest: "/var/tmp/{{ oraclejdk_file }}"
+  when: USE_PRIVATE_ORACLEJDK is undefined or not USE_PRIVATE_ORACLEJDK
+
+- name: Download Oracle Java from private S3 bucket
```

Comment:
@arjunsinghy96, my only concern here is that this is too specific (using S3 in a very specific way) and that we will have trouble upstreaming it as-is.

CC @0x29a, @giovannicimolin as the devops expert.

Reply:
@lgp171188 I totally agree with you here. The current core already allows replacing the download URL through extra variables, and this is a very specific way of retrieving a file.

I'd go for generating a new expiring URL for JDK and just setting `oraclejdk_url` in the extra settings instead of modifying the playbooks.